core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -253,6 +253,38 @@ object UnitAutomation {
         unit.movement.headTowards(encampmentToHeadTowards)
         return true
     }
+    
+    private fun tryRetreat(unit: MapUnit): Boolean {
```

Comment:
TrySwapRetreat, since this is only for swap

Reply:
Okay, I only included swapping because it can be done without much of a performance impact. If we want to change it to include non-swap tiles, we should move getDistanceToClosestEnemy() to be associated with the tiles instead of the units.